### PR TITLE
Data 653 integrate templates repository to ci

### PR DESCRIPTION
**Pull request by AdfPizarro**
Add github actions file to run rspec tests on the CI and fix the test

Notes: The gem will return the top level category template if the specific subcategory template is missing, the test for this feature relies on specific empty categories that were filled overtime, those tests were moved to a dummy category that won't be displayed on crowdcontrol

Jira ticket
[DATA-653](https://bugcrowd.atlassian.net/jira/software/projects/DATA/boards/81?assignee=62544befcdc24000704a2834&selectedIssue=DATA-653)

### CODE REVIEW

File: spec/bugcrowd_templates/template_path_spec.rb [hunk 1]
```diff
@@ -128,7 +128,7 @@
       let!(:subcategory) { 'clickjacking' }
       let!(:item) { 'form_input' }
       let!(:file_name) { 'template' }
-      let!(:result) { directory.join(type, field, category, subcategory, file_name_with_extension) }
+      let!(:result) { directory.join(type, field, category, subcategory, item, file_name_with_extension) }
```
[the remark this answers is not in the document]
Comment (AdfPizarro) [reply spec/bugcrowd_templates/template_path_spec.rb:131]: The item the templates is used to describe different template options for the same category, usually separated depending on how critical is the vulnerability is not exactly a subcategory (maybe that's the reason for the name "item") but is follows the same structure in the path 

`category/subcategory/item/template.md`

For exmple

`server_security_misconfiguration/directory_listing_enabled/`
This category > subcategory contains two folders (items) with a template
`non_sensitive_data_exposure`
`sensitive_data_exposure`

File: submissions/description/dummy_category/template.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# Dummy Category
```
[the remark this answers is not in the document]
Comment (AdfPizarro) [reply submissions/description/dummy_category/template.md]: The templates were displayed on cc based on the VRT i checked that and none of the dummy classes were displayed but yea you're right is better to have it in a fixture

Move dummy templates to the spec fixture folder
[b3164cb](https://github.com/bugcrowd/templates/pull/412/commits/b3164cb7b188a28262e26f9d9392226e5adebf2b)

File: spec/bugcrowd_templates/template_path_spec.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (AdfPizarro) [reply spec/bugcrowd_templates/template_path_spec.rb]: you're right that test was only for the other spec

Created dummy folders to test the file search on missing item and sub category templates, add test for both cases
no item template, return subcategory template
no subcategory template, return category template
Commit
https://github.com/bugcrowd/templates/pull/412/commits/1b7e6b5c29cac68c62639aa8e6fcc48796feebe4

File: spec/bugcrowd_templates_spec.rb [hunk 3]
```diff
@@ -196,7 +196,10 @@
         end
 
         context 'when there is no template in category folder' do
-          let!(:category) { 'using_components_with_known_vulnerabilities' }
+              let!(:directory) { Pathname.new(Gem::Specification.find_by_name('bugcrowd_templates').gem_dir)
+              .join('spec')
+              .join('fixture') }
+          let!(:category) { 'dummy_empty_category' }
```
[the remark this answers is not in the document]
Comment (AdfPizarro) [reply spec/bugcrowd_templates_spec.rb]: Done

https://github.com/bugcrowd/templates/pull/412/commits/4de93a6eefaa7bd6ce66013f5697ef410dd077da

File: spec/bugcrowd_templates_spec.rb [hunk 4]
```diff
@@ -196,7 +196,10 @@
         end
 
         context 'when there is no template in category folder' do
-          let!(:category) { 'using_components_with_known_vulnerabilities' }
+              let!(:directory) { Pathname.new(Gem::Specification.find_by_name('bugcrowd_templates').gem_dir)
```
[the remark this answers is not in the document]
Comment (AdfPizarro) [reply spec/bugcrowd_templates_spec.rb]: my bad, forgot to remove that line

Fixed
[4de93a6](https://github.com/bugcrowd/templates/pull/412/commits/4de93a6eefaa7bd6ce66013f5697ef410dd077da)

File: spec/bugcrowd_templates/template_path_spec.rb [hunk 5]
```diff
@@ -124,17 +124,41 @@
...
       let!(:field) { 'description' }
-      let!(:category) { 'server_security_misconfiguration' }
-      let!(:subcategory) { 'clickjacking' }
+      let!(:category) { 'fixture_category_for_items' }
+      let!(:subcategory) { 'fixture_subcategory' }
       let!(:item) { 'form_input' }
       let!(:file_name) { 'template' }
-      let!(:result) { directory.join(type, field, category, subcategory, file_name_with_extension) }
+      let!(:result) {
+        Pathname.new(Gem::Specification.find_by_name('bugcrowd_templates').gem_dir).join('spec', 'fixture')
```
[the remark this answers is not in the document]
Comment (AdfPizarro) [reply spec/bugcrowd_templates/template_path_spec.rb]: Done

[51e3f9a](https://github.com/bugcrowd/templates/pull/412/commits/51e3f9a4848f9887a56b84c8aeb7f7c85264d202)

File: spec/bugcrowd_templates_spec.rb [hunk 6]
```diff
@@ -196,7 +196,7 @@
         end
 
         context 'when there is no template in category folder' do
-          let!(:category) { 'using_components_with_known_vulnerabilities' }
+          let!(:category) { 'fixture_empty_category' }
```
[the remark this answers is not in the document]
Comment (AdfPizarro) [reply spec/bugcrowd_templates_spec.rb:203]: done 

[f29ddfd](https://github.com/bugcrowd/templates/pull/412/commits/f29ddfd22297637acc90e85991682a1cdaf598c3)

File: .github/workflows/rspec.yml [hunk 7]
```diff
@@ -0,0 +1,26 @@
+name: RSpec Tests
+
+on: [push, pull_request]
```
[the remark this answers is not in the document]
Comment (AdfPizarro) [reply .github/workflows/rspec.yml]: done

[5fc7e1c](https://github.com/bugcrowd/templates/pull/412/commits/5fc7e1ca3c20a92e9014623cac01cd7f23b34c1b)
